### PR TITLE
Correctly handle sequences with the same citations but in a different order

### LINKED ISSUE #5: Wrong output when the same refs in another place have another order
Suppose we have:

```
Correct: [@epstein_this_2023; @Kim2017]

Wrong: [@Kim2017; @epstein_this_2023]
```

It produces the following result (here `href`s are used as visual aids):

SCR-20240721-kuxi

As we can see, in both cases, the output is the same `[1, 2]`. But in the second case, it's obviously wrong.

### PR DESCRIPTION
**Pull request by DmitrySharabin**
Fixes #5

### CODE REVIEW

File: src/Bibliography.js [hunk 1]
```diff
@@ -138,11 +138,14 @@ export default class Bibliography {
 
 		let parts = parseFormattedCitationSequence(text, citations);
```
Comment (LeaVerou) [inline src/Bibliography.js]: But now we don't need to run this first — we can simply branch.

Comment (DmitrySharabin) [reply src/Bibliography.js]: Good point! Thanks

Comment (DmitrySharabin) [reply src/Bibliography.js]: Done! Did I get your idea right?

File: src/Bibliography.js [hunk 2]
```diff
@@ -136,13 +136,15 @@ export default class Bibliography {
 			reference.citations.push(uuid);
 		}
 
-		let parts = parseFormattedCitationSequence(text, citations);
-
-		if (parts === null) {
+		let parts;
+		if (citations.length > 1) {
 			// Sequence could not be parsed into parts directly
```
Comment (LeaVerou) [inline src/Bibliography.js]:
```suggestion
			// Sequence cannot be parsed into parts directly
```

File: src/Bibliography.js [hunk 3]
```diff
@@ -136,13 +136,15 @@ export default class Bibliography {
...
 			// Because not all citations are present in the output
```
Comment (LeaVerou) [inline src/Bibliography.js]: Ok this needs actual rewording not just a one word fix

Comment (DmitrySharabin) [reply src/Bibliography.js]: I tried to fix it. Could you please take another look?

**Review by LeaVerou — APPROVED** [review]

**Pull request merged**